riptide-spring-boot-autoconfigure/src/main/java/org/zalando/riptide/autoconfigure/RiptidePostProcessor.java

```diff
@@ -34,13 +35,13 @@ public void setEnvironment(final Environment environment) {
 
     @Override
     public void postProcessBeanDefinitionRegistry(final BeanDefinitionRegistry registry) {
-        final RiptideRegistrar registrar = registrarFactory.apply(new Registry(registry), Defaulting.withDefaults(properties));
-        registrar.register();
+        this.registry = registry;
     }
 
     @Override
     public void postProcessBeanFactory(final ConfigurableListableBeanFactory beanFactory) throws BeansException {
```

Comment:
Interesting that this seems to work since the javadoc for `postProcessBeanFactory` actually state:

> All bean definitions will have been loaded, but no beans will have been instantiated yet. This allows for overriding or adding properties even to eager-initializing beans.

To me that reads like the only thing left to do is modifying properties, but not register new bean definitions.

Reply:
At least reference doesn't state anything about prohibition of beans creation at this point. Spring it self using casting from BeanDefinitionRegistry to ConfigurableListableBeanFactory in postProcessBeanDefinitionRegistry, but not sure if it worth to go this way. Another idea would be perform class path scanning, but that's seems to be overkill

Reply:
Using bean factory was my initial idea but it didn't offer any ways to register new beans (definitions). It didn't occur to me that you can work with both in conjunction.